### PR TITLE
[SharpDX.Direct3D12] TextureCopyLocation alignment fix

### PR DESCRIPTION
**Pull request by mrvux**
Hello, this one was also missing in latest build and had explicit layout (but has a pointer to a resource)

Removed fields from mapping and added them in partial class instead

Thanks
Julien

### CODE REVIEW

**Comment by mrvux** [comment]
Ok since pushing my branch of course updated my pull request, a few additions (and as usual I noticed I forgot to push to master_integration...)

Overloads for cpu/gpu handles (to allow offsets)
Added GetMessage on infoqueue (same as old dx11 counterpart)
Updated mappings/functions to have working stream output description when creating PSO.
Small naming on Indirect argument description

File: Source/SharpDX.Direct3D12/Mapping.xml [hunk 1]
```diff
@@ -149,9 +156,19 @@
...
+    <map struct="D3D12_INDIRECT_ARGUMENT_DESC_INNER_0_INNER_2" name="ConstantBufferArgument"/>
+    <map struct="D3D12_INDIRECT_ARGUMENT_DESC_INNER_0_INNER_3" name="ShaderResourceViewArgument"/>
+    <map struct="D3D12_INDIRECT_ARGUMENT_DESC_INNER_0_INNER_4" name="UnorderedAccessViewArgument"/>
+    <move struct="D3D12_INDIRECT_ARGUMENT_DESC_INNER_(.*)" to="D3D12_INDIRECT_ARGUMENT_DESC"/>
+    
+    
     <!--Remove all fields for D3D12_RESOURCE_BARRIER_DESC as the union with pointers cannot be mapped safely in x86/x64 with C# explicit layout-->
     <remove field="D3D12_RESOURCE_BARRIER::.*"/>
     <remove field="D3D12_ROOT_PARAMETER::.*"/>
+    <remove field="D3D12_TEXTURE_COPY_LOCATION::.*" />
```
Comment (xoofx) [inline Source/SharpDX.Direct3D12/Mapping.xml]: In this case, we don't want to remove the TextureCopyLocation but instead rename the field to "ResourcePointer", leave it internal and provide a setter only for the pResource as it is done for example for [ComputePipelineStateDescription](https://github.com/sharpdx/SharpDX/blob/master/Source/SharpDX.Direct3D12/ComputePipelineStateDescription.cs#L35)

**Comment by mrvux** [comment]
Ok I updated Pull request, as my initial TextureCopyLocation layout was not correct anyway.

Problem is it can't be modelled after ComputePipelineStateDescription since it's also a Union type, similar to ResourceBarrier (and contains a resource pointer right at the beginning of the struct, this seems undetected by the code generator as it does generate an explicit layout).

I updated the struct so it works correctly, using a Union type as per ResourceBarrier.

**Comment by xoofx** [comment]
Thanks! it is merged in latest commit.

**Pull request closed without merging**